### ISSUE TITLE
issue.

### ISSUE THREAD

**Issue by mhertz** [emericg/OpenSubtitlesDownload#27]
I believe i've found an issue.

When running 'OpenSubtitlesDownload.py -l dan,swe,eng foo.avi'(or 'OpenSubtitlesDownload.py -l dan foo.avi' - btw, I believe the -a switch don't have any effect when running from the CLI, so I omit it), then a danish sub is downloaded, which doesn't match the file(sub is for ep2 but file is ep1). I then thought it was because of the filename searching heuristics which obviously cannot be foolproof always, but then I renamed the file from it's original filename, to instead 'test.avi' and ran the exact same line, and then the correct sub was downloaded in danish. So a sub is available with matching hash, but for some reason a filename-search is done instead. I also tried disabling filename-matching completely and ran again but with same issue.

I've tried this with three different files now(out of 15 or so), where renaming the file away from it's original name, so forcing the hash-check to work, fixed it.

If you could please kindly test with one of my files when you have the time, then I would much appreciate it, thanks.

[FILE]

Unrelated, but sometimes the app fails to connect and shows an error, and then when run a second time it works, or sometimes a third time is needed. Of course this is due to server-overload, but as it fails already after 1 second or so, then I was thinking if it possibly could be extended to retry the connection for a bit longer, or track that it error'ed out and retry automatically? It's not a big deal when running manually, but when having it set to run like me automatically after every torrent download, then it's more an issue there(as I don't know if a missing sub means no subs available or a server-overload). Well, I guess I could in my postproc.sh script check for error-code, if provided, and set to call the app again in those occasions, and once more for the few times, it takes 3 runs to connect... Anyway, just a thought when i'm reporting to you anyway :)

-Martin.

**Comment by mhertz** [comment]
Hmm, just wanted to add that after beeing consistently wrong on some files unless renaming them, then now it's behaving randomly???

Check out this screenshot, with the same command run 3 times, where one returns danish sub, the other an error(though not the normal server-overload), and the third a danish sub:
![screenshot](https://user-images.githubusercontent.com/378240/29324822-cb300554-81e5-11e7-9627-fc0f13304dea.png)

Note, the download part looks a little different than normal because I have changed wget to curl just for the downloading part and added a silent switch, since arch linux by default features curl instead of wget and as it was an easy change then I changed your script instead of installing wget(yes, i'm a little strange that way, I know :) )

I have of course tested first that my error wasen't related to that, and here's a screenshot showing the error with your unmodified version, except changing lang-suffix to off instead of auto:
![screenshot2](https://user-images.githubusercontent.com/378240/29325663-d7397efe-81e8-11e7-9576-5610bee9c93a.png)

**Comment by mhertz** [comment]
I just wanted to add that I checked the opensubtitles hash of both the file in question, and the file of the wrong sub retrieved(so S01E01 and S01E02). They had same file-size, so I was thinking maybe they would have same hash too, but they haden't. One was '6bbbe330768c5821' and the other '625738d2b4b3d774'. However, when doing the very advanced search with only the hash on opensubtitles.org, then both files are under each hash. I always thought that since hashes used filesize + first/last frames, then there couldn't be these mismatches, but what do I know.

Anyway, so it's not a bug persay, but I still think that it's strange that simply renaming a file, will change which sub is retrieved, every single time? I also checked another opensubtitles-auto-retriever(osdb lua-script for mpv) and there the first sub retrieved didn't change after renaming the file in question.

Just wanted to add that for completeness :) (and for others possibly reading this from a google-search)
